From 772a8fdbb86905795ef0a2882820aea948d1d302 Mon Sep 17 00:00:00 2001
Message-ID: <772a8fdbb86905795ef0a2882820aea948d1d302.1729453324.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1729453324.git.sharlatanus@gmail.com>
References: <cover.1729453324.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 10 Oct 2024 11:38:27 +0100
Subject: [PATCH 21/42] gnu: python-sep-pjw: Update to 1.3.6.

* gnu/packages/astronomy.scm (python-sep-pjw): Update to 1.3.6.
[phases]{relax-requirements}: Add more relaxation on not yet supported
option.
[native-inputs]: Add python-setuptools, python-setuptools-scm, and
python-wheel.

Change-Id: I507ca06def6afaaed70f9b31bb7a960437ba33d0
---
 gnu/packages/astronomy.scm | 25 ++++++++++++++++---------
 1 file changed, 16 insertions(+), 9 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 2bd1e834cb..ff9e6becac 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -5460,26 +5460,33 @@ (define-public python-sep
 (define-public python-sep-pjw
   (package
     (name "python-sep-pjw")
-    (version "1.3.5")
+    (version "1.3.6")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "sep_pjw" version))
        (sha256
-        (base32 "15jf16zycs1gz6jfkhmj7b8wdcpp8d5ikz15pmfkwq32a8mfdv8m"))))
+        (base32 "0ji3khgswidnnqx34js9fyk39kgh4gvf7q8gk2f7r4acaabzja2v"))))
     (build-system pyproject-build-system)
     (arguments
-     (list #:test-flags #~(list "test.py")
+     (list
+      #:test-flags #~(list "test.py")
       #:phases
       #~(modify-phases %standard-phases
-         (add-after 'unpack 'relax-requirements
-           (lambda _
-             (substitute* "pyproject.toml"
-               ;; numpy>=1.23.5
-               (("1.23.5") "1.23.2")))))))
+          (add-after 'unpack 'relax-requirements
+            (lambda _
+              (substitute* "pyproject.toml"
+                ;; numpy>=1.23.5
+                (("1.23.5") "1.23.2")
+                ;; TypeError: Configuration.__init__() got an unexpected
+                ;; keyword argument 'version_file'
+                (("version_file = .*") "")))))))
     (native-inputs
      (list python-cython
-           python-pytest))
+           python-pytest
+           python-setuptools
+           python-setuptools-scm
+           python-wheel))
     (propagated-inputs
      (list python-numpy))
     (home-page "https://github.com/PJ-Watson/sep-pjw")
-- 
2.46.0

